website/project/views/register.py

```diff
@@ -151,10 +447,42 @@ def node_register_template_page_post(auth, node, **kwargs):
     schema = MetaSchema.find(
         Q('name', 'eq', template)
     ).sort('-schema_version')[0]
+
+    # Create the registration
     register = node.register_node(
         schema, auth, template, json.dumps(clean_data),
     )
 
+    if data['registrationChoice'] == 'Make registration public immediately':
+        register.is_public = True
+        register.save()
+    elif data['registrationChoice'] == 'Enter registration into embargo':
+        # Sanitize embargo end date
+        embargo_end_date = datetime.datetime.strptime(
+            data['embargoEndDate'],
+            "%a, %d %b %Y %H:%M:%S %Z"
+        )
+
+        # Initiate embargo
+        try:
+            register.embargo_registration(auth.user, embargo_end_date)
+            register.add_log(
```

Comment:
Logging should not happen in the views; that is the responsibility of the model. Can this call be moved to `embargo_registration`?